monashspa/PHS2061/PHS2061_Q3_Fourier_Analysis.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ------------------------------------------------------------------------------
+# Q3: Fourier Analysis
+# Part 1: analysing a sound file
+# ------------------------------------------------------------------------------
+
+# Import the relevant libraries
+import monashspa
+import matplotlib.pyplot as plt
+import numpy as np
+import wave
+from matplotlib.widgets import Cursor
+
+# ------------------------------------------------------------------------------
+# First we have to import our recorded sample and extract the waveform 
+# ------------------------------------------------------------------------------
+
+## Choose a file to plot
+filename = "PHS2061_Q3_Guitar_Pluck.wav"
+
+## Open the file
+signal_wave = wave.open(filename, 'r')
+
+## Extract the sample rate from the file
+sample_rate = signal_wave.getframerate()
+
+## Create a numpy array from the waveform
+sig = np.frombuffer(signal_wave.readframes(-1), dtype=np.int16)
+
+## The audio is recorded in stereo so we want to
+## split the left and right channels apart.
+## We will only use the left channel from here onwards
+left, right = sig[0::2], sig[1::2]
+
+
+# ------------------------------------------------------------------------------
+# Now we can plot our sample. 
+# Try looking at different ranges by changing the start and end points.
+# ------------------------------------------------------------------------------
+
+## If you want to trim the sample 
+## choose suitable beginning and end points
+
+# sample_start  = 0
+# sample_end    = len(left)
+sample_start  = 90000
+sample_end    = 200000
+
+## Create the values for the x axis
+sample_numbers = np.arange(len(left))
+
+## The variables we want to plot on our graph
+x = sample_numbers[sample_start:sample_end]
+y = left[sample_start:sample_end]
+
+## Create a plot for the left channel 
+fig, (ax0, ax1) = plt.subplots(1, 2)
+ax0.plot(x,y)
+ax0.set_xlabel('Sample number')
+ax0.set_ylabel('Amplitude')
+ax0.set_title('Left channel')
+
+## Create a plot for the right channel
+x = sample_numbers[sample_start:sample_end]
+y = right[sample_start:sample_end]
+ax1.plot(x,y)
+ax1.set_xlabel('Sample number')
+ax1.set_title('Right channel')
+
+fig.suptitle('Recorded waveforms')
+
+plt.savefig('part1_waveform.pdf')
```

Comment:
The logbooks are written in Google Docs. I think it would be preferable to save figures in PNG format.

Reply:
Thanks for pointing this out, fixed